### ISSUE TITLE
Should we flag "unused" PEP-695 TypeVars?

### ISSUE THREAD

**Issue by AlexWaygood** [PyCQA/flake8-pyi#399]
Currently, if an old-style TypeVar with a private name is defined in a file but not used in that file, we emit Y018 to tell the user that the TypeVar is unused. Would it be worth extending this to PEP-695 TypeVars? We could potentially do something like this:

```py
type X[T] = list[T] | T  # Okay, the TypeVar T is used in the type alias
type Y[T] = int | str    # Y018: the TypeVar T isn't used in the type alias

class Foo[T]:            # Okay, the TypeVar T is used in the class body
    def meth(self) -> T: ...

class Bar[T]:            # Y018: the TypeVar T isn't used in the class body
    def meth(self) -> int: ...
```

Not sure if it would be worth doing this or not -- it might not add as much value as the current places we emit Y018, might be more complex to implement, and might have a higher chance of false positives.

Thoughts?

**Comment by hauntsaninja** [comment]
Seems like it could be a useful lint. Might be worth making a new error code though, since we could imagine issuing a similar warning for `class Baz(Generic[T])` that does not use T

**Comment by JelleZijlstra** [comment]
This should be an error, but I'm not sure we need flake8-pyi to implement it; I strongly suspect pyright will already give an error for such cases.

Definitely agree this should be a separate error code.

**Comment by AlexWaygood** [comment]
> This should be an error, but I'm not sure we need flake8-pyi to implement it; I strongly suspect pyright will already give an error for such cases.

That makes sense. I'll go ahead and close this for now; we can always reconsider once PEP-695 is more widely used, if it feels like it might be useful.
